source/resampling_with_code2.Rmd

```diff
@@ -0,0 +1,669 @@
+
+Now we can put these parts together to do 100 simulated trials:
+
+```{python, opts.label="py_ed"}
+# Procedure for 100 simulated trials.
+
+# An array to store the counts for each trial.
+z = np.zeros(100)
+
+# Repeat the trial procedure 100 times.
+for i in np.arange(0, 100):
+    rnd = np.random.default_rng()
+    # Get 12 random numbers from 0 through 99
+    zero_through_99 = np.arange(0, 100)
+    a = rnd.choice(zero_through_99, 12)
+    # How many numbers are greater than 74?
+    b = np.sum(a > 74)
+    # Store the result at the next position in the "z" array.
+    z[i] = b
+    # Now go back and do the next trial until finished.
+# Show the result of all 100 trials.
+z
+```
+
+```{r, opts.label="r_ed"}
+# Procedure for 100 simulated trials.
+
+# An array to store the counts for each trial.
+z <- numeric(100)
+
+# Repeat the trial procedure 100 times.
+for (i in 1:100) {
+    # Get 12 random numbers from 0 through 99
+    a <- sample(0:99, 12, replace=TRUE)
+    # How many are greater than 74?
+    b <- sum(a == 9)
+    # Store the result at the next position in the "z" array.
+    z[i] = b
+    # Now go back and do the next trial until finished.
+}
+# Show the result of all 100 trials.
+z
+```
+
+Finally, we need to count how many of the trials in `z` ended up with all-White juries.  These are the trials with a `z` value of 0.
+
+We can ask this question of *all 100 counts* by asking the question: is the {{< var array_name >}} `z` equal to 0, like this:
+
+```{python, opts.label="py_ed"}
+# Is the value of z equal to 0?
+all_white = z == 0
+# Show the result of the comparison.
+all_white
+```
+
+```{r, opts.label="r_ed"}
+# Is the value of z equal to 0?
+all_white <- z == 0
+# Show the result of the comparison.
+all_white
+```
+
+Now we need to get the number of
+[`True`]{.python}[`TRUE`]{.r}
+values in `all_white`, to find how many simulated trials gave all-white
+juries.
+
+```{python, opts.label="py_ed"}
+# Count the number of True values in "all_white"
+# This is the same as the number of values in "z" that are equal to 0.
+n_all_white = np.sum(all_white)
+# Show the result of the comparison.
+n_all_white
+```
+
+```{r, opts.label="r_ed"}
+# Count the number of True values in "all_white"
+# This is the same as the number of values in "z" that are equal to 0.
+n_all_white = sum(all_white)
+# Show the result of the comparison.
+n_all_white
+```
+
+`n_all_white` is the number of simulated trials for which all the jury members
+were White. It only remains to get the proportion of trials for which this was
+true, and to do this, we divide by the number of trials.
+
+```{python, opts.label="py_ed"}
+# Proportion of trials where all jury members were White.
+p = n_all_white / 100
+# Show the result
+p
+```
+
+```{r, opts.label="r_ed"}
+# Proportion of trials where all jury members were White.
+p <- n_all_white / 100
+# Show the result
+p
+```
+
+From this simulation experiment, we see that there is roughly a 2% chance that
```

Comment:
Maybe should show the classic calculation would be `p = 0.74 ** 12`?

Here's the three liner:

```
out = []
for i in range(100000):
    out.append(np.all(rng.integers(0, 100, size=12) <= 73))
np.sum(out)/100000 * 100
```

Now, may be worth also showing:

```
rng.choice([0, 1], p=[0.74, 0.26], size=12)
```

For when you do not want to deal with figuring out how numbers map to integers (which you can't very easily when they're fractions).

Reply:
Well - that could could go in a footnote - but we've promised them we will not do any mathematics, and they so far have no reason to know the multiplication rule for probability.   And it needs some explanation too, as to why we can use simple multiplication here (the jurors are independent ...)

Can we leave the choice discussion until the next chapter?  Yes, it's nicer, but this is all part of the gentle on-ramp ..